lib/bap_types/bap_helpers.ml

```diff
@@ -520,7 +523,7 @@ module Simpl = struct
       | OR,x,y  when x = y -> x
       | XOR,x,y when x = y -> zero width
       | XOR,x,y when is0 x -> y
-      | XOR,x,y when is0 y -> y
+      | XOR,x,y when is0 y -> x
```

Comment:
nice catch!
we definitely need a test case that will catch such transformations. Probably, some automatic expression generator. We need to think about this.